tests/e2e/utils/infrastructure.py

```diff
@@ -210,3 +214,173 @@ def _find_assets(self, asset_template: str, account_id: str, region: str):
+        self.region = self.session.region_name
+
+    def create_lambda_functions(self, function_props: Optional[Dict] = None):
+        """Create Lambda functions available under handlers_dir
+
+        It creates CloudFormation Outputs for every function found in PascalCase. For example,
+        {handlers_dir}/basic_handler.py creates `BasicHandler` and `BasicHandlerArn` outputs.
+
+
+        Parameters
+        ----------
+        function_props: Optional[Dict]
+            CDK Lambda FunctionProps as dictionary to override defaults
+
+        Examples
+        --------
+
+        Creating Lambda functions available in the handlers directory
+
+        ```python
+        self.create_lambda_functions()
+        ```
+
+        Creating Lambda functions and override runtime to Python 3.7
+
+        ```python
+        from aws_cdk.aws_lambda import Runtime
+
+        self.create_lambda_functions(function_props={"runtime": Runtime.PYTHON_3_7)
+        ```
+        """
+        handlers = list(self.handlers_dir.rglob("*.py"))
+        source = Code.from_asset(f"{self.handlers_dir}")
+        props_override = function_props or {}
+
+        for fn in handlers:
+            fn_name = fn.stem
+            function_settings = {
+                "id": f"{fn_name}-lambda",
+                "code": source,
+                "handler": f"{fn_name}.lambda_handler",
+                "tracing": Tracing.ACTIVE,
+                "runtime": Runtime.PYTHON_3_9,
+                "layers": [
+                    LayerVersion.from_layer_version_arn(
+                        self.stack,
+                        f"{fn_name}-lambda-powertools",
+                        f"arn:aws:lambda:{self.region}:017000801446:layer:AWSLambdaPowertoolsPython:29",
+                    )
+                ],
+                **props_override,
+            }
+
+            function_python = Function(self.stack, **function_settings)
+
+            aws_logs.LogGroup(
+                self.stack,
+                id=f"{fn_name}-lg",
+                log_group_name=f"/aws/lambda/{function_python.function_name}",
+                retention=aws_logs.RetentionDays.ONE_DAY,
+                removal_policy=RemovalPolicy.DESTROY,
+            )
+
+            # CFN Outputs only support hyphen
+            fn_name_pascal_case = fn_name.title().replace("_", "")  # basic_handler -> BasicHandler
+            self._add_resource_output(
+                name=fn_name_pascal_case, value=function_python.function_name, arn=function_python.function_arn
+            )
+
+    def deploy(self) -> Dict[str, str]:
+        """Creates CloudFormation Stack and return stack outputs as dict
+
+        Returns
+        -------
+        Dict[str, str]
+            CloudFormation Stack Outputs with output key and value
+        """
+        template, asset_manifest_file = self._synthesize()
+        assets = Assets(cfn_template=asset_manifest_file, account_id=self.account_id, region=self.region)
```

Comment:
I think that asset_manifest_file is not a cfn template. It is separate entity. I would suggest renaming this variable to `asset_manifest_file` or something similar

Reply:
Great catch!! Renaming to `asset_manifest`.